### PR TITLE
Integrate Vuex in todo app

### PR DESCRIPTION
**Pull request by PranjayPathak**
- Added Vuex to the todo app

### CODE REVIEW

File: src/components/TaskCell.vue [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      this.$store.dispatch("deleteTask", this.task.id);
+      // this.$emit("delTask", this.task.id);
+    },
+    editTask() {
+      this.$emit("editTask", this.task.id);
+    },
+    toggleStatus() {
+      this.$store.dispatch("toggleStatus", this.task.id);
+      // this.$emit("toggleStatus", this.task.id);
+    },
```
Comment (ankitboghra) [inline src/components/TaskCell.vue]: why not emit event here also, same as above?

File: src/components/TaskCell.vue [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+<template>
+  <div class="task_container">
+    <div class="flex_container p-2 my-2 rounded-sm bg-primary-200">
+      <div class="flex_container">
+        <input
+          class="taskcell_checkbox mx-2"
+          type="checkbox"
+          id="done"
+          value="true"
+          v-bind:checked="task.isCompleted"
```
Comment (ankitboghra) [inline src/components/TaskCell.vue]: please use getters instead of destructuring the object here

File: src/store/index.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        data: "Sample task 4",
+        id: 1246702392149,
+        isCompleted: true,
+      },
+    ],
+  },
+  getters: {},
+  mutations: {
+    addTask: (state, newTask) => {
+      let check = true; //flag to check if task already exists
```
Comment (ankitboghra) [inline src/store/index.js]: checks should be kept in action
mutation should just simply update the data

**Review by ankitboghra — CHANGES_REQUESTED** [review]

File: src/App.vue [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    };
+  },
+  computed: {
+    demoTasks() {
+      return this.$store.state.demoTasks;
+    },
+  },
+  methods: {
+    addTask() {
+      this.$store.dispatch("addTask", this.newTask);
```
Comment (ujjwalgupta94) [inline src/App.vue:15]: use `mapAction`

File: src/App.vue [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+import TaskCell from "./components/TaskCell.vue";
+export default {
+  name: "App",
+  data() {
+    return {
+      newTask: "",
+    };
+  },
+  computed: {
+    demoTasks() {
```
Comment (ujjwalgupta94) [inline src/App.vue]: use `mapState`

**Review by ujjwalgupta94 — CHANGES_REQUESTED** [review]

**Pull request merged**